runtime/src/test/scala/com/soundcloud/twinagle/ServerEndpointBuilderSpec.scala

```diff
@@ -0,0 +1,30 @@
+package com.soundcloud.twinagle
+
+import com.soundcloud.twinagle.test.TestMessage
+import com.twitter.finagle.Filter
+import com.twitter.finagle.http.{MediaType, Request, Status}
+import com.twitter.util.{Await, Future, Throw}
+import org.specs2.mutable.Specification
+import org.specs2.specification.Scope
+
+class ServerEndpointBuilderSpec extends Specification {
+
+  "catches exceptions thrown by user code" in {
+    def oops(x: TestMessage): Future[TestMessage] = {
+      throw new Exception("oops")
+    }
+
+    val ex = new Exception("oh dear")
+    val builder = new ServerEndpointBuilder(_ => Filter.TypeAgnostic.Identity)
+    val (_, svc) = builder.build(
+      oops,
+      EndpointMetadata("/twirp", "service", "rpc")
+    )
+
+    val request = Request()
+    request.mediaType = MediaType.Json
+    request.contentString = "{}"
+
+    Await.result(svc(request).liftToTry) must not throwAn(ex)
```

Comment:
How about `Await.result(svc(request).liftToTry).isThrow ==== true`? Then we don't need the awkward `throwAn(ex)` matcher and also ensure that the exception is actually wrapped inside the Future.

Reply:
ah whoops. `ex` is a leftover from a refactoring. I agree it's confusing this way